bin/crew

```diff
@@ -1964,10 +1838,9 @@ end
 def upgrade_command(args) = upgrade(*args['<name>'], build_from_source: @opt_source)
 
 def upload_command(args)
-  upload if args['<name>'].empty?
   args['<name>'].each do |name|
-    search name
-    upload name if @pkg.name
+    search name, pkg_path: File.join(CREW_LOCAL_REPO_ROOT, 'packages', "#{name}.rb")
```

Comment:
No need to pass the package path.  Let the function figure it out from the predefined constant.  Also, we do not want to use `CREW_LOCAL_REPO_ROOT`.  We should be using `CREW_PACKAGES_PATH`.

Reply:
You have to copy the package from your local repo to the crew packages directory BEFORE you can even test it so I'm not sure why you think this needs a refactor.  Also, please stop resolving conversations.

Reply:
This is NOT my workflow.  This is harder than a simple `crew build libnvme` and NOT requiring the path.  The path is abstracted which is the whole point of why I wrote this originally.

Reply:
> This is NOT my workflow.

I'm not centering my refactor around your personal workflow.

Reply:
Obviously because you don't understand my workflow.  I'm trying to make it EASIER, not HARDER, which is apparently the route you want to take.

Reply:
`crew` already requires that you be in a local repo-- I don't understand why it would be easier to then edit the actual crew installation?

> Obviously because you don't understand my workflow.

Apparently not. Would you mind explaining it so I can ensure it continues working?

Reply:
I'm abstracting the path so if you are making changes to the package in your local repo, crew will detect that it differs from the crew repo package and offer to copy it over so it can run it inside crew.  So simply running `crew build/install/reinstall <package>` will enable you to test the package without needed to copy the package file over manually.  My vision is for contributors to be able to do everything they need to do to create and update packages inside their own forked repo instead of working inside crew directly.  This way changes are easier to manage and removes the potential human error element from the process.  I know @satmandu has a different workflow and that is fine but I still think mine is easier. :)

Reply:
So your goal is that the user does all the changes in their local repository, but crew then copies it to the installation, works with that, then copies back any changes to the local repository?

Reply:
I'll point out that the workflow I am enabling here does not require any manual copying, and in fact just removes the requirement for a package file to be inside the crew installation. (I'm also trying to reach an agreement on this because it will significantly affect how I refactor `crew check`, which I'd like to get started on soon)

Reply:
I still don't think you fully understand my workflow.  The way it currently works (by design) is any change you make in your local working repo will be prompted to copy to crew for installation (no manual copying).  I don't think a refactor is necessary because it will over complicate the process and the package has to end up in crew anyway to test the install.  You can always do a crew update to restore back to the original version if you are updating existing packages.

Reply:
The point of these changes to `search` is to ensure that there is no need for any copying, whether automatic or manual. I'd argue that this simplifies the process, in fact.